examples/acados_python/pendulum_on_cart/solution_sensitivities/sensitivity_utils.py

```diff
@@ -83,25 +84,35 @@ def export_parametric_ocp(
 
     ocp.solver_options.N_horizon = N_horizon
 
+    # set mass to one
+    ocp.p_global_values = np.ones((1,))
+
     Q_mat = 2 * np.diag([1e3, 1e3, 1e-2, 1e-2])
     R_mat = 2 * np.diag([1e-1])
 
     ocp.cost.cost_type = "EXTERNAL"
     ocp.cost.cost_type_e = "EXTERNAL"
 
+    if cost_scale_as_param:
+        # add parameter to model
+        cost_scale_param = ca.SX.sym('cost_scale_param')
+        ocp.model.p_global = ca.vertcat(ocp.model.p_global, cost_scale_param)
+        ocp.p_global_values = np.concatenate((ocp.p_global_values, np.ones((1,))))
+        # add nonlinear dependency in cost
+        cost_scale_factor = ca.exp(cost_scale_param)
+    else:
+        cost_scale_factor = ca.exp(ocp.model.p_global)
```

Comment:
don't quite get what you want to do/test here? would expect the the cost_scale_factor to be constant in the else part?

Reply:
I think it is mainly to test with more than 1 parameters. Additionally, it allows to make the cost (non)linear in the parameter which is good for testing.
But yes, in `else` it should be a fixed value. Just pushed a fix.